api/src/services/fields.ts

```diff
@@ -834,7 +834,11 @@ export class FieldsService {
 				column = table.increments(field.field);
 			}
 		} else if (field.type === 'string') {
-			column = table.string(field.field, field.schema?.max_length ?? undefined);
+			if (field.schema?.data_type && field.schema.max_length) {
+				column = table.specificType(field.field, `${field.schema.data_type}(${field.schema.max_length})`);
```

Comment:
This feels like an opportunity for a sql injection attack, even if only admins can run an import here.

Reply:
A lot worse can be done if are able to inject sql here, so not a real concern in my opinion.

Reply:
There's an elevation difference between Directus admins and sysadmins who have direct SQL or server host access. This would expose SQL injection from the Directus admin level which would elevate their access to sysadmin. That's also why we can't just run raw SQL or code from flows for example 🙂